internal/archive/zip/zip.go

```diff
@@ -55,11 +56,14 @@ func (w *Writer) Add(src, dst string) (int, error) {
 		if err != nil {
 			return 0, err
 		}
-		b, err := os.ReadFile(src)
+		f, err := os.Open(src)
 		if err != nil {
 			return 0, err
 		}
-		_, err = w.Write(b)
+		defer f.Close()
+
+		reader := bufio.NewReader(f)
+		_, err = reader.WriteTo(w)
```

Comment:
I think this can be simplified to
```suggestion
		_, err = io.Copy(w, f)
```

Underneath, the `Copy` function will also create a buffer to do the copying :)